hermes/src/network/wormhole.rs

```diff
@@ -0,0 +1,340 @@
+}
+
+impl std::fmt::Display for GuardianSet {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "[")?;
+        for (i, key) in self.keys.iter().enumerate() {
+            // Comma seperated printing of the keys using hex encoding.
+            if i != 0 {
+                write!(f, ", ")?;
+            }
+
+            write!(f, "{}", hex::encode(key))?;
+        }
+        write!(f, "]")
+    }
+}
+
+/// BridgeData extracted from wormhole bridge account, due to no API.
+#[derive(borsh::BorshDeserialize)]
+#[allow(dead_code)]
+pub struct BridgeData {
+    pub guardian_set_index: u32,
+    pub last_lamports:      u64,
+    pub config:             BridgeConfig,
+}
+
+/// BridgeConfig extracted from wormhole bridge account, due to no API.
+#[derive(borsh::BorshDeserialize)]
+#[allow(dead_code)]
+pub struct BridgeConfig {
+    pub guardian_set_expiration_time: u32,
+    pub fee:                          u64,
+}
+
+/// GuardianSetData extracted from wormhole bridge account, due to no API.
+#[derive(borsh::BorshDeserialize)]
+pub struct GuardianSetData {
+    pub index:           u32,
+    pub keys:            Vec<[u8; 20]>,
+    pub creation_time:   u32,
+    pub expiration_time: u32,
+}
+
+/// Update the guardian set with the given ID in the state.
+pub async fn update_guardian_set(state: &State, id: u32, guardian_set: GuardianSet) {
+    let mut guardian_sets = state.guardian_set.write().await;
+    guardian_sets.insert(id, guardian_set);
+}
+
+/// Wormhole `prost` compiled definitions.
+///
+/// We use `prost` to build the protobuf definitions from the upstream Wormhole repository. Which
+/// outputs `.rs` files during execution of build.rs, these can be included into the source while
+/// compilation is happening.
+///
+/// The following module structure must match the protobuf definitions, so that the generated code
+/// can correctly reference modules from each other.
+mod proto {
+    pub mod node {
+        pub mod v1 {
+            include!(concat!(env!("OUT_DIR"), "/node.v1.rs"));
+        }
+    }
+
+    pub mod gossip {
+        pub mod v1 {
+            include!(concat!(env!("OUT_DIR"), "/gossip.v1.rs"));
+        }
+    }
+
+    pub mod spy {
+        pub mod v1 {
+            include!(concat!(env!("OUT_DIR"), "/spy.v1.rs"));
+        }
+    }
+
+    pub mod publicrpc {
+        pub mod v1 {
+            include!(concat!(env!("OUT_DIR"), "/publicrpc.v1.rs"));
+        }
+    }
+}
+
+// Launches the Wormhole gRPC service.
+pub async fn spawn(state: Arc<State>) -> Result<()> {
+    while !crate::SHOULD_EXIT.load(Ordering::Acquire) {
+        if let Err(e) = run(state.clone()).await {
+            tracing::error!(error = ?e, "Wormhole gRPC service failed.");
+        }
+
+        tokio::time::sleep(tokio::time::Duration::from_secs(5)).await;
+    }
+    Ok(())
+}
+
+async fn run(state: Arc<State>) -> Result<()> {
+    let mut client = SpyRpcServiceClient::connect("grpc://127.0.0.1:7073").await?;
```

Comment:
Maybe pass the address as a value from the outside?

Reply:
That was the intent I missed this, will fix.